### PR TITLE
Acceptance Tests

### PR DESCRIPTION
**Pull request by ishustava**
This PR proposes to merge the work from `acceptance-tests-base` feature branch to master.

All future acceptance tests work will come in as a PR directly to master.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -175,3 +222,152 @@ Here are some examples of common test patterns:
...
+
+If you are adding a feature that fits thematically with one of the existing test suites,
+then you need to add your test cases to the existing test files.
+Otherwise, you will need to create a new test suite.
+
+We recommend to start by either copying the [example test](test/acceptance/tests/example/example_test.go)
+or the whole [example test suite](test/acceptance/tests/example),
+depending on the test you need to add.
+
+#### Adding Test Suites
```
Comment (ashwin-venkatesh) [inline CONTRIBUTING.md:236]: ❤️

Comment (ashwin-venkatesh) [reply CONTRIBUTING.md:236]: This entire section is documentation gold.

**Review by lkysow — APPROVED** [review]

**Review by thisisnotashwin — APPROVED** [review]
This PR is 🥇

File: CONTRIBUTING.md (hunk 1 above)
Comment (ishustava) [reply CONTRIBUTING.md:236]: thanks!

**Pull request merged**